Shared/Samples/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,273 @@
+        /// protocol string.
+        /// - Returns: A tuple of the accessory and its protocol,
+        ///            or nil if no supported accessory exists.
+        func firstSupportedAccessoryWithProtocol() -> (EAAccessory, String)? {
+            for accessory in EAAccessoryManager.shared().connectedAccessories {
+                // The protocol string to establish the EASession.
+                guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
+                    // Skip the accessories with protocol not for NMEA data transfer.
+                    continue
+                }
+                // Only return the first connected and supported accessory.
+                return (accessory, protocolString)
+            }
+            return nil
+        }
+        
+        /// The Bluetooth accessory picker connected to a supported accessory.
+        func accessoryDidConnect(connectedAccessory: EAAccessory, protocolString: String) {
+            if let dataSource = NMEALocationDataSource(accessory: connectedAccessory, protocol: protocolString) {
+                nmeaLocationDataSource = dataSource
+                start()
+            }
+        }
+        
+        /// Reset the sample, stops the data source, cancels tasks, and resets button states and status strings.
+        func reset() {
+            // Reset buttons states.
+            isResetButtonDisabled = true
+            isSourceButtonDisabled = false
+            
+            // Reset the status text.
+            accuracyStatus = "Accuracy info will be shown here."
+            satelliteStatus = "Satellites info will be shown here."
+            
+            // Reset NMEA location data source.
+            nmeaLocationDataSource = nil
+            autoPanMode = .off
+
+            Task {
+                // Stop the location display, which in turn stop the data source.
+                await locationDisplay.dataSource.stop()
+            }
+            
+            // Cancel the autoPan, location, and satellite observation tasks.
+            cancelObservationTasks()
+            
+            // Pause the mock data generation.
+            mockNMEADataSource.stop()
+        }
+
+        /// Starts the location data source and awaits location and satellite updates.
+        /// - Parameter usingMockedData: Indicates that the location datasource should use mocked data.
+        func start(usingMockedData: Bool = false) {
+            // Set NMEA location data source for location display.
+            locationDisplay.dataSource = nmeaLocationDataSource
+            // Set buttons states.
+            isSourceButtonDisabled = true
+            isResetButtonDisabled = false
+            
+            // Start the data source and location display.
+            if usingMockedData {
+                mockNMEADataSource.start(with: nmeaLocationDataSource)
+            }
```

Comment:
If the previous suggestion is applied, I think the logic here would be

If mock data is used…
1. initialize an empty `NMEALocationDataSource`
2. create an async task to receive data from the async sequence from the `mockNMEADataSource`

If real device is used…
1. initialize a `NMEALocationDataSource` using the `NMEALocationDataSource.init(accessory:protocol:)` initializer

Therefore, the data source is always initialized in the model, not in the view.